packages/console/src/onboarding/App.tsx

```diff
@@ -37,6 +44,28 @@ function App() {
     };
   }, [setThemeOverride]);
 
+  /**
+   * This `useEffect()` initiates Google Tag and report a sign-up conversion to it.
+   * It may run multiple times (e.g. a user visit multiple times to finish the onboarding process,
+   * which rarely happens), but it'll be okay since we've set conversion's "Count" to "One"
+   * which means only the first interaction is valuable.
+   *
+   * Track this conversion in the backend has been considered, but Google does not provide
+   * a clear guideline for it and marks the [Node library](https://developers.google.com/tag-platform/tag-manager/api/v2/libraries)
+   * as "alpha" which looks unreliable.
+   */
+  useEffect(() => {
+    if (shouldReportToGtag) {
+      gtag('js', new Date());
+      gtag('config', gtagAwTrackingId);
+      gtag('event', 'conversion', {
+        send_to: gtagSignUpConversionId,
+      });
+    } else {
+      console.debug('Google Tag event fires');
```

Comment:
Just want to confirm this is expected.

Reply:
good call. i did it intentionally by now i think this should be moved into the if clause